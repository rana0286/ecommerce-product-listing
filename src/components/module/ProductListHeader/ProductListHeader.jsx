import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ProductsSizeFilter, ProductsListHeading }from '../../blocks';
import { Header } from '../../elements';
import noop from 'lodash/noop';
import './ProductListHeader.scss';

class ProductListHeader extends PureComponent {
  render() {
    return (<Header>
      <div className="product-list-header__heading">
        <ProductsListHeading headingLabel={this.props.headingLabel} />
      </div>
      <div className="product-list-header__filter">
        <ProductsSizeFilter onSelect={this.props.onSelect} options={this.props.productSizeOptions} />
      </div>
    </Header>);
  }
}

ProductListHeader.propTypes = {
  headingLabel: PropTypes.string,
  onSelect: PropTypes.func,
  productSizeOptions: PropTypes.arrayOf(PropTypes.object),
};

ProductListHeader.defaultValue = {
  headingLabel: '',
  onSelect: noop,
  productSizeOptions: [],
};

export default ProductListHeader;
