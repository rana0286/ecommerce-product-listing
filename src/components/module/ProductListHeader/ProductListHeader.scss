.product-list-header__heading {
  flex-grow: 1;
}

.product-list-header__filter {
  flex-grow: 1;
}
