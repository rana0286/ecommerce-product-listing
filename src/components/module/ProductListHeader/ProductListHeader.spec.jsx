import React from 'react';
import ProductListHeader from './ProductListHeader';
import renderer from 'react-test-renderer';

describe('src/components/module/ProductListHeader', () => {
  it('ProductListHeader component should render', () => {
    const tree = renderer.create(<ProductListHeader
      headingLabel="Test label"
      onSelect={jest.fn()}
      productSizeOptions={[{value:'X', label: 'X'}]}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
