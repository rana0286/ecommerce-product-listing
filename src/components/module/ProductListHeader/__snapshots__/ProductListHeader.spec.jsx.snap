// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/module/ProductListHeader ProductListHeader component should render 1`] = `
<div
  className="product-list-header"
>
  <div
    className="product-list-header__heading"
  >
    <h2
      className="heading heading--h2"
    >
      Test label
    </h2>
  </div>
  <div
    className="product-list-header__filter"
  >
    <div
      className="select-box "
    >
      <label
        className="select-box__label"
        htmlFor="select-box"
      >
        
      </label>
      <select
        className="select-box__select"
        id="select-box"
        onChange={[Function]}
        onSelect={[MockFunction]}
        selected=""
        value=""
      >
        <option
          value="X"
        >
          X
        </option>
      </select>
    </div>
  </div>
</div>
`;
